packages/agent-toolkit/src/core/tools/platform-api-tools/read-doc-tool.ts

```diff
@@ -0,0 +1,161 @@
+
+  protected async executeInternal(input: ToolInputType<typeof readDocToolSchema>): Promise<ToolOutputType<never>> {
+    // Validate that at least one filter is provided
+    if (!input.ids && !input.object_ids && !input.workspace_ids) {
+      return {
+        content:
+          'Error: You must provide at least one filter: ids, object_ids, or workspace_ids to search for documents.',
+      };
+    }
+
+    try {
+      const variables: ReadDocsQueryVariables = {
+        ids: input.ids,
+        object_ids: input.object_ids,
+        limit: input.limit || 25,
+        order_by: input.order_by,
+        page: input.page,
+        workspace_ids: input.workspace_ids,
+      };
+
+      let res = await this.mondayApi.request<ReadDocsQuery>(readDocs, variables);
+
+      // If no results found and ids were provided, try treating the ids as object_ids - sometimes the user inputs ids are actually object_ids
+      if ((!res.docs || res.docs.length === 0) && input.ids) {
+        const fallbackVariables: ReadDocsQueryVariables = {
+          ids: undefined,
+          object_ids: input.ids, // Try the provided ids as object_ids
+          limit: input.limit || 25,
+          order_by: input.order_by,
+          page: input.page,
+          workspace_ids: input.workspace_ids,
+        };
+
+        res = await this.mondayApi.request<ReadDocsQuery>(readDocs, fallbackVariables);
+      }
+
+      if (!res.docs || res.docs.length === 0) {
+        return {
+          content: 'No documents found matching the specified criteria.',
+        };
+      }
+
+      return await this.enrichDocsWithMarkdown(res.docs);
+    } catch (error) {
+      return {
+        content: `Error reading documents: ${error instanceof Error ? error.message : 'Unknown error occurred'}`,
+      };
+    }
+  }
+
+  // Convert docs content to markdown string
+  private async enrichDocsWithMarkdown(docs: NonNullable<ReadDocsQuery['docs']>): Promise<ToolOutputType<never>> {
+    const docsInfo = await Promise.all(
+      docs.map(async (doc) => {
+        // Get markdown content for this doc
+        let blocks_as_markdown = '';
+        try {
+          const markdownVariables: ExportMarkdownFromDocMutationVariables = {
+            docId: doc!.id,
+          };
+
+          const markdownRes = await this.mondayApi.request<ExportMarkdownFromDocMutation>(
+            exportMarkdownFromDoc,
+            markdownVariables,
+          );
+
+          if (markdownRes.export_markdown_from_doc.success && markdownRes.export_markdown_from_doc.markdown) {
+            blocks_as_markdown = markdownRes.export_markdown_from_doc.markdown;
+          } else {
+            blocks_as_markdown = `Error getting markdown: ${markdownRes.export_markdown_from_doc.error || 'Unknown error'}`;
+          }
+        } catch (error) {
+          blocks_as_markdown = `Error getting markdown: ${error instanceof Error ? error.message : 'Unknown error'}`;
+        }
+
+        return {
+          id: doc!.id,
```

Comment:
You don't need to do it if you define the doc as non nullable in the original type in the map